src/api/Tidy5eSheetsApi.ts

```diff
@@ -1,86 +1,146 @@
+      return;
+    }
 
-  unregisterCharacterSheetTab(tabId: string) {
-    return unregisterCharacterSheetTab(tabId);
+    CharacterSheetRuntime.registerTab(registeredTab);
   }
 
-  registerNpcSheetTab(
-    tab: SheetTabState<NpcSheetContext>,
-    options?: SheetTabRegistrationOptions
-  ) {
-    return registerNpcSheetTab(tab, options);
-  }
+  /**
+   * Adds a tab to the available NPC sheet tabs.
+   * @param tab the information necessary to render a tab
+   * @param layout an optional sheet layout or layouts (default: 'all')
+   * @returns void
+   */
+  registerNpcTab(
+    tab: HandlebarsTab | HtmlTab,
+    layout?: SheetLayout | SheetLayout[]
+  ): void {
+    if (!ActorSheetRuntimeManager.validateTab(tab)) {
+      return;
+    }
+    const registeredTab = ActorSheetRuntimeManager.mapCustomTabToRegisteredTab(
+      tab,
+      layout
+    );
+
+    if (!registeredTab) {
+      warn('Unable to register tab. Tab type not supported');
+      return;
+    }
 
-  unregisterNpcSheetTab(tabId: string) {
-    return unregisterNpcSheetTab(tabId);
+    NpcSheetRuntime.registerTab(registeredTab);
   }
 
-  registerVehicleSheetTab(
-    tab: SheetTabState<VehicleSheetContext>,
-    options?: SheetTabRegistrationOptions
-  ) {
-    return registerVehicleSheetTab(tab, options);
+  /**
+   * Adds a tab to the available Vehicle sheet tabs.
+   * @param tab the information necessary to render a tab
+   * @param layout an optional sheet layout or layouts (default: 'all')
+   * @returns void
+   */
+  registerVehicleTab(
+    tab: HandlebarsTab | HtmlTab,
+    layout?: SheetLayout | SheetLayout[]
+  ): void {
+    if (!ActorSheetRuntimeManager.validateTab(tab)) {
+      return;
+    }
+    const registeredTab = ActorSheetRuntimeManager.mapCustomTabToRegisteredTab(
+      tab,
+      layout
+    );
+
+    if (!registeredTab) {
+      warn('Unable to register tab. Tab type not supported');
+      return;
+    }
+
+    VehicleSheetRuntime.registerTab(registeredTab);
   }
 
-  unregisterVehicleSheetTab(tabId: string) {
-    return unregisterVehicleSheetTab(tabId);
+  /**
+   * Adds a tab to all relevant item sheets.
+   * @see {@link CustomTabBase} for options related to all tabs.
+   * @param tab the custom tab settings to use when incorporating this tab.
+   * @example Register an item tab for spell items only, adding some custom data to the Item Sheet Context object before rendering my handlebars template
+   * ```js
+   * Hooks.once("tidy5e-sheet.ready", (api) => {
+   *   api.registerItemTab(
+   *     new api.models.HandlebarsTab({
+   *       title: "My Item Tab",
+   *       path: "/modules/my-module/my-item-tab.hbs",
+   *       enabled: (data) => data.item.type === 'spell',
+   *       getData: (data) => {
+   *         data['my-extra-data'] = "Hello, world! 👋";
+   *         return data;
+   *       }
+   *     }));
+   * });
+   * ```
+   */
+  registerItemTab(tab: HandlebarsTab | HtmlTab): void {
+    ItemSheetRuntime.registerTab(tab);
```

Comment:
Separate the API type from the Sheet Runtime type. This is to prevent accidental API breakage from renaming fields.